### PR TITLE
Task 5.1

### PR DESCRIPTION
**Pull request by fujoshik**

### CODE REVIEW

File: Task 5.1/Task5.1/SparseMatrix.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
+﻿using System.Collections;
+
+namespace Task5._1
+{
```
Comment (IgorKheidorov) [inline Task 5.1/Task5.1/SparseMatrix.cs:6]: Not effective, this is not a sparse matrix. You allocate memory for long[rows, columns], to store only 2 non-zero values, for example. Please think to store only non-zero values inside, and provide them by indexer (idea is similar to Diagonal matrix)

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]
Please see comment and redo the task

File: Task 5.1/Task5.1/SparseMatrix.cs (hunk 1 above)
Comment (fujoshik) [reply Task 5.1/Task5.1/SparseMatrix.cs:6]: Fixed

File: Task 5.1/Task5.1/SparseMatrix.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        }
+
+        public IEnumerable<(int,int,long)> GetNonzeroElements()
+        {
+            var result = new List<(int,int,long)>();
+            for (int i = 0; i < Columns; i++)
+            {
+                for (int j = 0; j < Rows; j++)
+                {
+                    if (_elements[new Tuple<int, int>(j, i)] != 0)
```
Comment (IgorKheidorov) [inline Task 5.1/Task5.1/SparseMatrix.cs]: better use yield return not creating a List

File: Task 5.1/Task5.1/SparseMatrix.cs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        public int GetCount(int x)
+        {
+            int count = 0;
```
Comment (IgorKheidorov) [inline Task 5.1/Task5.1/SparseMatrix.cs]: 1. Use LINQ here
2. What about GetCount(0)?

File: Task 5.1/Task5.1/SparseMatrix.cs [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                return _elements.ContainsKey(key) ? _elements[key] : 0;
+            }
+            set
+            {
+                if (i < 0 || j < 0 || i >= Rows || j >= Columns)
+                {
+                    throw new IndexOutOfRangeException("Indexes must be greater than 0 and less than the size of the matrix");
+                }
+
+                var key = new Tuple<int, int>(i, j);
```
Comment (IgorKheidorov) [inline Task 5.1/Task5.1/SparseMatrix.cs:46]: new value can be 0, in this case you need to remove it from matrix

File: Task 5.1/Task5.1/SparseMatrix.cs [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+                }
+                else
+                {
+                    _elements.Add(key, value);
+                }
+            }
+        }
+
+        public override string ToString()
+        {
```
Comment (IgorKheidorov) [inline Task 5.1/Task5.1/SparseMatrix.cs:67]: Use StringBuilder to concatenate strings

**Review by IgorKheidorov — APPROVED** [review]
Please fix comments and can merge

**Pull request merged**